xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRoom.java

```diff
@@ -1353,6 +1353,8 @@ public CompletableFuture<Void> clearChatHistory() {
             MUCPersistenceManager.clearRoomChatFromDB(this);
             // Remove the history of the room from memory (preventing it to pop up in a new room by the same name).
             roomHistory.purge();
+            // Fire event to clear chat room history
+            MUCEventDispatcher.roomClearChatHistory(getSelfRepresentation().getOccupantJID());
```

Comment:
Why `getSelfRepresentation().getOccupantJID()`? Isn't a simple `getJID()` enough?

Reply:
MUCEventDispatcher calls often uses getSelfRepresentation().getOccupantJID() in the existing code, but I have changed MUCEventDispatcher.roomClearChatHistory and MUCEventDispatcher.roomDestroyed to use getJID() as suggested.